meteor/lib/api/rest.ts

```diff
@@ -0,0 +1,204 @@
+	/**
+	 * Returns the current version of Sofie
+	 */
+	index(): Promise<ClientAPI.ClientResponse<{ version: string }>>
+	/**
+	 * Activates a Playlist.
+	 *
+	 * Throws if there is already an active Playlist for the studio that the Playlist belongs to.
+	 * @param rundownPlaylistId Playlist to activate.
+	 * @param rehearsal Whether to activate into rehearsal mode.
+	 * @param connection Connection data including client and header details
+	 */
+	activate(
+		rundownPlaylistId: RundownPlaylistId,
+		rehearsal: boolean,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Deactivates a Playlist.
+	 *
+	 * Throws if the Playlist is not currently active.
+	 * @param rundownPlaylistId Playlist to deactivate.
+	 * @param connection Connection data including client and header details
+	 */
+	deactivate(
+		rundownPlaylistId: RundownPlaylistId,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<void>>
+	/**
+	 * Executes the requested action by passing `actionId` directly to the blueprints `executeAction` function.
+	 *
+	 * Throws if the target Playlist is not currently active.
+	 * Throws if there is not an on-air Part.
+	 * Throws if the `executeAction` method of the blueprints throws - blueprint error will be returned to caller.
+	 * @returns An object in the form {queuedPartInstanceId?: PartInstanceId, taken?: boolean} (if a part was queued by the action/if the next part was automatically taken)
+	 * @param rundownPlaylistId Playlist to execute action in.
+	 * @param actionId Action Id string, should match something inside the blueprints.
+	 * @param userData Any value, recommended to be a relatively simple { [key: string]: string }-style object, as the object may need to survive a round of serialization.
+	 * @param connection Connection data including client and header details
+	 */
+	executeAction(
+		rundownPlaylistId: RundownPlaylistId,
+		actionId: string,
+		userData: any,
+		connection: Meteor.Connection
+	): Promise<ClientAPI.ClientResponse<object>>
+	/**
+	 * Executes the requested AdLib/AdLib Action. This is a "planned" AdLib (Action) that has been produced by the blueprints during the ingest process.
+	 *
+	 * Throws if the target Playlist is not active.
+	 * Throws if there is not an on-at part instance.
+	 * @returns a `ClientResponseError` if an adLib for the provided `adLibId` cannot be found.
+	 * @param rundownPlaylistId Playlist to execute adLib in.
+	 * @param adLibId AdLib to execute.
+	 * @param connection Connection data including client and header details
+	 * @param triggerMode A string to specify a particular variation for the AdLibAction, valid actionType strings are to be read from the status API.
+	 */
+	executeAdLib(
+		rundownPlaylistId: RundownPlaylistId,
+		adLibId: AdLibActionId | RundownBaselineAdLibActionId | PieceId | BucketAdLibId,
+		connection: Meteor.Connection,
+		triggerMode?: string
+	): Promise<ClientAPI.ClientResponse<object>>
```

Comment:
This doesn't allow triggering some "AdLib-like" items that are present in the GUI, specifically: "Sticky" source-layer adLibs and "Clear" source-layer adLibs.

Reply:
I wonder how to represent those. Perhaps `clearLayer` and `recallSticky` or similar new methods could be exposed to abstract away the fact that those are adlibs? Becuase isn't the adlib representation of those actions sort of a cheat to get those actions to appear in various UIs?

Reply:
I've added methods for clearing sourcelayers / recalling sticky Pieces https://github.com/nrkno/sofie-core/pull/777/commits/57c07b14883054732eeec54fb137f6b125d652dc